src/Microsoft.OData.Client/DataServiceClientFormat.cs

```diff
@@ -275,4 +283,89 @@ private static string ChooseMediaType(bool hasSelectQueryOption)
             return MimeApplicationJsonODataLight;
         }
     }
+
+    /// <summary>
+    ///This class is a utility to help with POCO loading it can be registered as an option to load the CSDL over network or from a file or string
+    ///
+    /// </summary>
+    public class DataServiceClientFormatUtil
+    {
+        public IEdmModel EdmModel { get; private set; }
+        /// <summary>
+        /// Calling this initializes the parsing in the constructor
+        /// </summary>
+        /// <param name="metadataUri"></param>
+        public DataServiceClientFormatUtil(Uri metadataUri)
+        {
+            EdmModel = ReadModelFromNetwork(metadataUri, new Dictionary<string, string>());
+        }
+        public DataServiceClientFormatUtil(Uri metadataUri, IDictionary<string, string> headers)
+        {
+            EdmModel = ReadModelFromNetwork(metadataUri, headers);
+        }
+        /// <summary>
+        /// Reads the model from network
+        /// </summary>
+        /// <param name="metadataUri"></param>
+        /// <returns></returns>
+        private static IEdmModel ReadModelFromNetwork(Uri metadataUri, IDictionary<string, string> headers)
+        {
+            Task<string> response = FetchXML(metadataUri, headers);
+            string strModel = response.Result;
+            return ParseModel(strModel);
+        }
+
+        private static Task<string>  FetchXML(Uri metadataUri, IDictionary<string, string> headers)
+        {
+            HttpClient client = new HttpClient();
+            Task<HttpResponseMessage> response = null;
+            HttpRequestMessage metadataRequestMessage = new HttpRequestMessage(HttpMethod.Get, metadataUri);
+            try
+            {
+                foreach (KeyValuePair<string, string> header in headers)
+                {
+                    metadataRequestMessage.Headers.Add(header.Key, header.Value);
+                }
+
+                response = client.SendAsync(metadataRequestMessage);
+                return response.Result.Content.ReadAsStringAsync();
+            }
+            finally
+            {
+                if (response != null)
+                    response.Dispose();
+
+                metadataRequestMessage.Dispose();
+                client.Dispose();
+            }
+        }
+
+        private static IEdmModel ParseModel(string strModel)

```

Comment:
why don't you call the method in Edm lib?

Reply:
Done